auth0_flutter/test/web/auth0_flutter_web_test.dart

```diff
@@ -72,13 +74,91 @@ void main() {
 
   test('handleRedirectCallback is called on load when auth params exist in URL',
       () async {
+    final interop.RedirectLoginResult mockRedirectResult =
+        interop.RedirectLoginResult();
+
+    when(mockClientProxy.isAuthenticated())
+        .thenAnswer((final _) => Future.value(false));
+    when(mockClientProxy.handleRedirectCallback())
+        .thenAnswer((final _) => Future.value(mockRedirectResult));
+
+    plugin.urlSearchProvider = () => '?code=abc&state=123';
+    await auth0.onLoad();
+    verify(mockClientProxy.handleRedirectCallback());
+    verifyNever(mockClientProxy.checkSession());
+  });
+
+  test('handleRedirectCallback captures appState that was passed', () async {
+    final Map<String, Object?> appState = <String, Object?>{
+      'someFancyState': 'value',
+    };
+
+    final interop.RedirectLoginResult mockRedirectResult =
+        interop.RedirectLoginResult(
+      appState: appState,
+    );
+
     when(mockClientProxy.isAuthenticated())
         .thenAnswer((final _) => Future.value(false));
+    when(mockClientProxy.handleRedirectCallback(any))
+        .thenAnswer((final _) => Future.value(mockRedirectResult));
 
     plugin.urlSearchProvider = () => '?code=abc&state=123';
     await auth0.onLoad();
     verify(mockClientProxy.handleRedirectCallback());
     verifyNever(mockClientProxy.checkSession());
+
+    final Object? capturedAppState = await auth0.appState;
+
+    expect(capturedAppState, isNotNull);
+    expect(capturedAppState, isA<Map<Object?, Object?>>());
+    capturedAppState as Map<Object?, Object?>;
+    const MapEquality<Object?, Object?> eq = MapEquality<Object?, Object?>();
+
+    expect(eq.equals(capturedAppState, appState), isTrue);
+  });
+
+  test('appState getter throws UnimplementedError when not overridden',
+      () async {
+    final Auth0FlutterWebPlatform webImpl = StubAuth0FlutterWeb();
+
+    expect(
+      () async => webImpl.appState,
+      throwsA(predicate((final Object? error) {
+        if (error is! UnimplementedError) {
+          return false;
+        }
+
+        return error.message == 'web.appState has not been implemented';
+      })),
+    );
+  });
+
+  test('appState getter returns value when accessed more than once', () async {
```

Comment:
This test was added to prevent regressing the behavior of the new getter. (it should always return the same value)